src/Neo/Wallets/Helper.cs

```diff
@@ -126,35 +128,76 @@ public static long CalculateNetworkFee(this Transaction tx, DataCache snapshot,
+                                    script.EmitPush(true);
+                                    break;
+                                case ContractParameterType.Integer:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    break;
+                                case ContractParameterType.Hash160:
+                                    script.EmitPush(new byte[UInt160.Length]);
+                                    break;
+                                case ContractParameterType.Hash256:
+                                    script.EmitPush(new byte[UInt256.Length]);
+                                    break;
+                                case ContractParameterType.PublicKey:
+                                    script.EmitPush(new byte[33]);
+                                    break;
+                                case ContractParameterType.Array:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    script.Emit(OpCode.NEWARRAY);
+                                    break;
+                                case ContractParameterType.Map:
+                                    script.Emit(OpCode.NEWMAP);
+                                    break;
+                                default:
+                                    throw new ArgumentException("The verify method requires parameters that need to be passed via the witness' invocation script.");
+                            }
+                        }
+                        invocationScript = script.ToArray();
+                    }
 
                     // Empty verification and non-empty invocation scripts
-                    var invSize = invocationScript?.GetVarSize() ?? Array.Empty<byte>().GetVarSize();
-                    size += Array.Empty<byte>().GetVarSize() + invSize;
+                    var invSize = invocationScript?.GetVarSize() ?? System.Array.Empty<byte>().GetVarSize();
+                    size += System.Array.Empty<byte>().GetVarSize() + invSize;
 
                     // Check verify cost
                     using ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, tx, snapshot.CreateSnapshot(), settings: settings, gas: maxExecutionCost);
                     engine.LoadContract(contract, md, CallFlags.ReadOnly);
                     if (invocationScript != null) engine.LoadScript(invocationScript, configureState: p => p.CallFlags = CallFlags.None);
-                    if (engine.Execute() == VMState.FAULT) throw new ArgumentException($"Smart contract {contract.Hash} verification fault.");
-                    if (!engine.ResultStack.Pop().GetBoolean()) throw new ArgumentException($"Smart contract {contract.Hash} returns false.");
+                    _ = engine.Execute(); // https://github.com/neo-project/neo/issues/2805
```

Comment:
I think we still need to handle non-boolean result, because if the result of invocation is not a bool, then something wrong happened during invocation, and thus user can't properly rely on the GAS consumed, so it's an error.

Reply:
If we check the boolean result, we need to check the FAULT state.

Reply:
@AnnaShaleva how can you trust in the result if you don't check the state?

Reply:
Check the state, if it's HALT, then check the stack. If the stack contains a single Boolean then accumulate GAS consumed and move on to the next witness. If the state is FAULT or the result is neither `true` nor `false` or more items are left on stack then return error to the user. Exactly like it's done here:
https://github.com/nspcc-dev/neo-go/blob/7304b2c7fbb802dac986596a82352ab98655eecd/pkg/services/rpcsrv/server.go#L993

Reply:
Done